apis/v1/shared_types.go

```diff
@@ -334,9 +334,10 @@ const (
 	RouteReasonNoMatchingListenerHostname RouteConditionReason = "NoMatchingListenerHostname"
 
 	// This reason is used with the "Accepted" condition when there are
-	// no matching Parents. In the case of Gateways, this can occur when
-	// a Route ParentRef specifies a Port and/or SectionName that does not
-	// match any Listeners in the Gateway.
+	// no matching Parents. In the case of Gateways, this can occur when:
+	// * no referenced parent exists
```

Comment:
If "no referenced parent exists", who would populate this status? It's possible for a controller to populate the status if a Gateway they're implementing is referenced, but just a nonexistent port or section name, but it seems impossible to know who should populate this if a nonexistent Gateway/parent resource is referenced here.

Reply:
At Kong (unmanaged gateway), we have a single ingress controller that watches all (or a subset of) namespaces and reconciles all `Gateway`s and all `*Route`s in those namespaces. In the contexts where there is no 1:1 (or n:1) relationship between gateway and gateway controller, but instead, a gateway controller reconciles everything, if no referenced gateway exists the unique gateway controller should be the one in charge of setting this status condition.

What is not immediately clear to me is: how this pattern is common among our implementations and if the Gateway API expects such a pattern.

Reply:
As written, Gateway API does not expect a pattern like that at all. It's expected that the only Gateway API objects touched by an implementation are those that are associated with a GatewayClass managed by the implementation - so there needs to be a tree of references that has a GatewayClass as a root node. Another thing we've never explicitly said in that way, but that's definitely what we intended.

This is the other reason why we're reluctant to allow people to skip having a GatewayClass object as well, but it seems clear that the only-have-the-implementation-watch-a-subset-of-namespaces use case needs to be addressed in some fashion - with recommendations if not specifications. I'm making a note about this one - I think that checking in with api-machinery about it may be essential.

Reply:
> there needs to be a tree of references that has a GatewayClass as a root node.

I see, this is a good way of putting it. What do you think about adding a conformance test for this? We could create a fake `GatewayClass` and a chain of objects having that GWC as root (`Gateway`, `HTTPRoute`) and check that all the created objects are ignored by the controller and no status gets updated.

Reply:
Hmm, that seems like a good idea to me.